zingo-testutils/src/assertions.rs

```diff
@@ -60,3 +60,20 @@ pub async fn assert_send_outputs_match_recipient<NoteId>(
         );
     }
 }
+
+/// assert that proposing shield causes a specific error
+pub async fn assert_cant_shield(client: &zingolib::lightclient::LightClient) {
+    assert!(match client.propose_shield().await.unwrap_err() {
+            zingolib::lightclient::propose::ProposeShieldError::Component(zcash_client_backend::data_api::error::Error::NoteSelection(zcash_client_backend::data_api::wallet::input_selection::GreedyInputSelectorError::Change(zcash_client_backend::fees::ChangeError::InsufficientFunds { available, required }))) => {
+                println!("available {}, required more than {}", available.into_u64(), required.into_u64());
+                true
+            },
+            zingolib::lightclient::propose::ProposeShieldError::BelowThreshold(amount) => {
+                println!("possible note {}, threshold {}", amount, zingolib::lightclient::propose::SHIELDING_CUTOFF);
+                true},
+            e => {
+                dbg!(e);
```

Comment:
i think we should use println! for test printouts that are meant to be permenant useful data when a test fails